src/GafferImage/FilterAlgo.cpp

```diff
@@ -0,0 +1,222 @@
+//////////////////////////////////////////////////////////////////////////
+////
+////  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+////
+////  Redistribution and use in source and binary forms, with or without
+////  modification, are permitted provided that the following conditions are
+////  met:
+////
+////      * Redistributions of source code must retain the above
+////        copyright notice, this list of conditions and the following
+////        disclaimer.
+////
+////      * Redistributions in binary form must reproduce the above
+////        copyright notice, this list of conditions and the following
+////        disclaimer in the documentation and/or other materials provided with
+////        the distribution.
+////
+////      * Neither the name of John Haddon nor the names of
+////        any other contributors to this software may be used to endorse or
+////        promote products derived from this software without specific prior
+////        written permission.
+////
+////  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+////  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+////  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+////  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+////  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+////  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+////  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+////  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+////  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+////  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+////  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+////
+//////////////////////////////////////////////////////////////////////////
+
+
+#include "GafferImage/FilterAlgo.h"
+#include "IECore/Exception.h"
+
+#include <climits>
+
+
+using namespace Imath;
+using namespace GafferImage;
+using namespace GafferImage::FilterAlgo;
+
+namespace
+{
+
+// OIIO's built in gaussian filter is clipped quite aggressively, so there
+// is a significant step to zero at the edge of the filter. This makes it
+// unsuitable for use in a gaussian blur - as the filter radius expands to
+// include another pixel, it switches on suddenly rather than fades up from
+// black. This variant approaches very close to zero at the edge of its
+// support, making it more suitable for use in a blur.
+class SmoothGaussian2D : public OIIO::Filter2D
```

Comment:
I see this is copied over from Resample, but that you haven't removed it from there. Can we rejig things so we don't have duplicates? Maybe we can make `FilterAlgo::createFilter()` usable from Resample too somehow? Perhaps keeping just the logic for upsize/downsize filter defaults in Resample?